packages/client/src/workflow-client.ts

```diff
@@ -946,6 +924,31 @@ export class WorkflowClient {
       followRuns: options?.followRuns ?? true,
     });
   }
+
+  /**
+   * List workflows by given `query`.
+   *
+   * See more info on how to build the query in the Temporal docs: https://docs.temporal.io/visibility
+   *
+   * This API requires Elasticsearch based visibility as of server version 1.18.
+   */
+  public async *list(query: string, options?: ListOptions): AsyncIterable<WorkflowExecutionInfo> {
+    let nextPageToken: Uint8Array = Buffer.alloc(0);
+    for (;;) {
+      const respose = await this.workflowService.listWorkflowExecutions({
+        namespace: this.options.namespace,
+        query,
+        nextPageToken,
+        pageSize: options?.pageSize,
+      });
+      // Not decoding memo payloads concurrently even though we could have to keep the lazy nature of this iterator.
```

Comment:
Any particular reason?

Reply:
It's not worth it IMHO for now, memos tend to generally be small.